artemis-journal/src/main/java/org/apache/activemq/artemis/core/io/nio/NIOSequentialFile.java

```diff
@@ -51,6 +55,34 @@ public NIOSequentialFile(final SequentialFileFactory factory,
       this.maxIO = maxIO;
    }
 
+   @Override
+   protected TimedBufferObserver createTimedBufferObserver() {
+      return new LocalBufferObserver() {
```

Comment:
Make this its own class, so its not anonymous.

Reply:
I can do it, but (for this specific class) I can't see the benefits.
In a future implementation I would like to abstract away things like `AbstractSequentialFile::fileSize` and create 2 separate classes:
`AsyncLocalBufferObserver` and `SyncLocalBufferObserver`.
The former that perform the copy because it assymes asynchrounous writes, the latter not.
But at this point, given that the classes are packed into their owner seq file class probably I could just let them be where/as they are to reduce the code changes.

Reply:
At least make it a nested class. Anonymous classes whilst not impossible to debug, just having it named makes stacks trace, heap dumps etc that little bit easier to quickly read and understand.

Reply:
Good point about stacks trace bud, will do it immediately!